src/drivers/distance_sensor/cm8jl65/cm8jl65.cpp

```diff
@@ -1,742 +0,0 @@
-	int _file_descriptor{-1};
-	int _orb_class_instance{-1};
-
-	uint8_t _cycle_counter{0};
-	uint8_t _linebuf[25] {};
-	uint8_t _rotation{0};
-
-	uint16_t _crc16{0};
-
-	// Use conservative distance bounds, to make sure we don't fuse garbage data
-	float _max_distance{7.9f}; // Datasheet: 8.0m
-	float _min_distance{0.2f}; // Datasheet: 0.17m
-
-	CM8JL65_PARSE_STATE _parse_state{WAITING_FRAME};
-
-	orb_advert_t _distance_sensor_topic{nullptr};
-
-	perf_counter_t _comms_errors{perf_alloc(PC_COUNT, "cm8jl65_com_err")};
-	perf_counter_t _sample_perf{perf_alloc(PC_ELAPSED, "cm8jl65_read")};
-};
-
-
-CM8JL65::CM8JL65(const char *port, uint8_t rotation) :
-	CDev(RANGE_FINDER_BASE_DEVICE_PATH),
-	ScheduledWorkItem(px4::wq_configurations::hp_default),
-	_rotation(rotation)
-{
-	// Store the port name.
-	strncpy(_port, port, sizeof(_port) - 1);
-
-	// Enforce null termination.
-	_port[sizeof(_port) - 1] = '\0';
-}
-
-CM8JL65::~CM8JL65()
-{
-	// Ensure we are truly inactive.
-	stop();
-
-	perf_free(_sample_perf);
-	perf_free(_comms_errors);
-}
-
-uint16_t
-CM8JL65::crc16_calc(const unsigned char *data_frame, uint8_t crc16_length)
-{
-	// compute CRC16 IBM 8005
-	unsigned char crc_high_byte = 0xFF;
-	unsigned char crc_low_byte = 0xFF;
-	size_t index = 0;
-
-	while (crc16_length--) {
-		index = crc_low_byte ^ *(data_frame++);
-		crc_low_byte = (unsigned char)(crc_high_byte ^ crc_msb_vector[index]);
-		crc_high_byte = crc_lsb_vector[index];
-	}
-
-	uint16_t crc16 = (crc_high_byte << 8 | crc_low_byte);
-	crc16 = (crc16 >> 8) | (crc16 << 8); // Convert endian
-
-	return crc16;
-}
-
-int
-CM8JL65::collect()
-{
-	perf_begin(_sample_perf);
-
-	int bytes_processed = 0;
-	int distance_mm = -1;
-	int index = 0;
-
-	bool crc_valid = false;
-
-	// Read from the sensor UART buffer.
-	int bytes_read = ::read(_file_descriptor, &_linebuf[0], sizeof(_linebuf));
-
-	if (bytes_read > 0) {
-		index = bytes_read - 6 ;
-
-		while (index >= 0 && !crc_valid) {
-			if (_linebuf[index] == START_FRAME_DIGIT1) {
-				bytes_processed = index;
-
-				while (bytes_processed < bytes_read && !crc_valid) {
-					if (data_parser(_linebuf[bytes_processed], _frame_data, _parse_state, _crc16, distance_mm) == PX4_OK) {
-						crc_valid = true;
-					}
-
-					bytes_processed++;
-				}
-
-				_parse_state = WAITING_FRAME;
-			}
-
-			index--;
-		}
-
-	} else {
-		PX4_INFO("read error: %d", bytes_read);
-		perf_count(_comms_errors);
-		perf_end(_sample_perf);
-		return PX4_ERROR;
-	}
-
-	if (!crc_valid) {
-		return -EAGAIN;
-	}
-
-	bytes_read = OK;
-
-	distance_sensor_s report = {};
-	report.current_distance = static_cast<float>(distance_mm) / 1000.0f;
-	report.id               = 0;	// TODO: set proper ID.
-	report.h_fov            = 0.0488692f;
-	report.max_distance     = _max_distance;
-	report.min_distance     = _min_distance;
-	report.orientation      = _rotation;
-	report.signal_quality   = report.current_distance < _max_distance && report.current_distance > _min_distance ? -1 : 0;
```

Comment:
I guess this logic should now be implemented somewhere around [here](https://github.com/PX4/Firmware/blob/6f3868b5ba75de38a409c2ad72251dfaaee99a3e/src/lib/drivers/rangefinder/PX4Rangefinder.cpp#L83) ?

Reply:
Yes, we could do this by default everywhere if a real signal_quality isn't provided.

Reply:
Added logic to PX4Rangefinder to set quality to 0 if otherwise unset (-1) and current_distance is outside of bounds.